src/common/Utility.js

```diff
@@ -420,6 +420,18 @@
                 }
                 return dest;
             }
+        },
+        stringify: function(obj) {
+            var cache = [];
```

Comment:
I'm not entirely following what you are doing here,  But from my Naive point perspective a few things to consider-- if it does what you need then I'm good with this.  
1. Since this cache is local to the function is it giving you any lift after 50% of your JSON is processed?
2. It seems the code is caching values disassociated with the keys - wouldn't it be reasonable to have same values under different keys.